### PR TITLE
fix: project_api_key data source missing project_assignment attribute

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Ticket: [INTMDB-969](https://jira.mongodb.org/browse/INTMDB-969)

This PR adds the `project_assignment` to the data source `project_api_key` has we forgot to add it

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Tested locally + Added acceptance tests

```bash
terraform state show data.mongodbatlas_project_api_key.test
# data.mongodbatlas_project_api_key.test:
data "mongodbatlas_project_api_key" "test" {
    api_key_id         = "64c7eb2c175e1e5d357fb05c"
    description        = "TF key"
    id                 = "terraform-20230731171109811200000001"
    private_key        = "********-****-****-56f6fb43a7b3"
    project_assignment = [
        {
            project_id = "63f4d4a47baeac59406dc131"
            role_names = [
                "GROUP_OWNER",
                "GROUP_READ_ONLY",
            ]
        },
        {
            project_id = "64b8d3219c4cdc3f9108e318"
            role_names = [
                "GROUP_READ_ONLY",
            ]
        },
    ]
    project_id         = "63f4d4a47baeac59406dc131"
    public_key         = "estakgih"
    role_names         = [
        "GROUP_OWNER",
        "GROUP_READ_ONLY",
    ]
}
```

```bash
terraform state show data.mongodbatlas_project_api_keys.test
# data.mongodbatlas_project_api_keys.test:
data "mongodbatlas_project_api_keys" "test" {
    id         = "terraform-20230731171830248700000001"
    project_id = "63f4d4a47baeac59406dc131"
    results    = [
        {
            api_key_id         = "64c7eb2c175e1e5d357fb05c"
            description        = "TF key"
            private_key        = "********-****-****-56f6fb43a7b3"
            project_assignment = [
                {
                    project_id = "63f4d4a47baeac59406dc131"
                    role_names = [
                        "GROUP_OWNER",
                        "GROUP_READ_ONLY",
                    ]
                },
                {
                    project_id = "64b8d3219c4cdc3f9108e318"
                    role_names = [
                        "GROUP_READ_ONLY",
                    ]
                },
            ]
            public_key         = "estakgih"
            role_names         = [
                "GROUP_OWNER",
                "GROUP_READ_ONLY",
            ]
        },
    ]
}

```

### CODE REVIEW

File: website/docs/d/project_api_key.html.markdown [hunk 1]
```diff
@@ -44,5 +44,11 @@ In addition to all arguments above, the following attributes are exported:
 * `public_key` - Public key for this Organization API key.
 * `private_key` - Private key for this Organization API key.
 * `role_names` - Name of the role. This resource returns all the roles the user has in Atlas. The [MongoDB Documentation](https://www.mongodb.com/docs/atlas/reference/user-roles/#project-roles) describes the valid roles that can be assigned.
+### project_assignment
```
Comment (Zuhairahmed) [inline website/docs/d/project_api_key.html.markdown:47]: This looks good, but do we need to make associated `project_assignment` doc edits to mongodbatlas_project_api_key**s** data source as well?

Comment (andreaangiolillo) [reply website/docs/d/project_api_key.html.markdown:47]: yes, good point. Thanks. I will update it

File: mongodbatlas/data_source_mongodbatlas_project_api_key.go [hunk 2]
```diff
@@ -39,6 +39,26 @@ func dataSourceMongoDBAtlasProjectAPIKey() *schema.Resource {
 				Elem: &schema.Schema{
 					Type: schema.TypeString,
 				},
+				Deprecated: fmt.Sprintf(DeprecationMessageParameterToResource, "v1.12.0", "project_assignment"),
```
Comment (AgustinBettati) [inline mongodbatlas/data_source_mongodbatlas_project_api_key.go:42]: Just for consistency, this flag can also be added in project api keys data source

**Review by AgustinBettati — CHANGES_REQUESTED** [review]

File: mongodbatlas/data_source_mongodbatlas_project_api_key.go (hunk 2 above)
Comment (andreaangiolillo) [reply mongodbatlas/data_source_mongodbatlas_project_api_key.go:42]: upadated

File: website/docs/d/project_api_key.html.markdown (hunk 1 above)
Comment (andreaangiolillo) [reply website/docs/d/project_api_key.html.markdown:47]: updated

**Review by AgustinBettati — APPROVED** [review]

**Review by Zuhairahmed — APPROVED** [review]
LGTM

**Pull request merged**